tests/e2e/taqueria-plugin-ipfs-pinata.spec.ts

```diff
@@ -1,132 +1,186 @@
+		expect(stderr).toEqual(expect.arrayContaining(['ipfs hash was not provided']));
+
+		await cleanup();
 	});
 
-	test('publish a directory to ipfs and get back the expected cid for each file', async () => {
-		const publishRun = await exec(`taq publish ./ipfs/`, { cwd: taqueriaProjectPath });
+	test('publish a file will return the expected cid', async () => {
+		const { execute, cleanup, exists, writeFile } = await prepareEnvironment();
+		const {} = await execute('taq', 'init test-project');
+		await exists('./test-project/.taq/config.json');
+
+		const env_file =
+			'pinataJwtToken=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySW5mb3JtYXRpb24iOnsiaWQiOiI1M2FkZjkxZi0yY2Q0LTQ3ZGUtYThlOS00YmM0YjI5NDI4NzYiLCJlbWFpbCI6Im1pY2hhZWxrZXJuYWdoYW5AZWNhZGxhYnMuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsInBpbl9wb2xpY3kiOnsicmVnaW9ucyI6W3siaWQiOiJOWUMxIiwiZGVzaXJlZFJlcGxpY2F0aW9uQ291bnQiOjF9XSwidmVyc2lvbiI6MX0sIm1mYV9lbmFibGVkIjpmYWxzZSwic3RhdHVzIjoiQUNUSVZFIn0sImF1dGhlbnRpY2F0aW9uVHlwZSI6InNjb3BlZEtleSIsInNjb3BlZEtleUtleSI6ImNiOGMzMzVkN2RhOThiZGQ2MTRmIiwic2NvcGVkS2V5U2VjcmV0IjoiZWIwYmUxYjRhYzNhZjE5ZGE3MjQ3MTAxNmFlZjFjNTllNjQzNTdlOTcwZmY1ZmYxZDBjNmU1ZTBkYmI1ODkzMCIsImlhdCI6MTY2OTEzMzI2MH0.U9JyTzEviH5y-GxSmD6YEkCVkcjsSX6d13a9fHE70LM';
+		await writeFile('./test-project/.env', env_file);
+
+		const {} = await execute('taq', 'install ../taqueria-plugin-ipfs-pinata', './test-project');
+		await exists('./test-project/node_modules/@taqueria/plugin-ipfs-pinata/index.js');
+
+		const ipfs_file = await (await exec('cat e2e/data/ipfs/0001.txt')).stdout;
+		await writeFile('./test-project/ipfs/0001.txt', ipfs_file);
 
-		for (let i = 1; i <= 12; i++) {
-			expect(publishRun.stdout).toContain(`${i}.txt`);
-		}
+		const { stdout: stdout1 } = await execute('taq', 'publish ipfs/0001.txt', './test-project');
+		expect(stdout1).toEqual(
+			expect.arrayContaining([
+				'│ ✔ │ {{base}}/test-project/ipfs/0001.txt │ QmZcEXgfE9K4zFVkTrqW5x4skvEqpNajZzvb7frrWnkQa5 │',
+			]),
+		);
 
-		expect(publishRun.stdout).toMatch(/0001\.txt.*QmZcEXgfE9K4zFVkTrqW5x4skvEqpNajZzvb7frrWnkQa5/i);
-		expect(publishRun.stdout).toMatch(/0005\.txt.*QmQZF8kEyU6iuyc4xbnkTYNTtMEL4gjYgXhPjcXK14gnJ1/i);
-		expect(publishRun.stdout).toMatch(/0010\.txt.*QmW6GqMj6bXPXygCVCeVvcZN6w1w8ou4aVNKvarQrz8N4w/i);
-		expect(publishRun.stdout).toMatch(/0012\.txt.*QmZLSfobCPA8rQrm6dss28gyDwqXaw8GhQ8CXXQMSMnuNm/i);
+		await cleanup();
 	});
 
-	test('pin using an ipfs hash', async () => {
-		const publishRun = await exec(`taq pin QmZcEXgfE9K4zFVkTrqW5x4skvEqpNajZzvb7frrWnkQa5`, {
-			cwd: taqueriaProjectPath,
-		});
-		expect(publishRun.stdout).toMatch(/QmZcEXgfE9K4zFVkTrqW5x4skvEqpNajZzvb7frrWnkQa5/i);
+	test('publish a directory will return the expected cid for each file', async () => {
+		const { execute, cleanup, exists, writeFile } = await prepareEnvironment();
+		const {} = await execute('taq', 'init test-project');
+		await exists('./test-project/.taq/config.json');
+
+		const env_file =
```

Comment:
If these are all the same tokens I would declare a const at the top of the file. Small fish.

Reply:
yeah in this case.  done